gateway-server/src/main/java/org/apache/knox/gateway/services/token/impl/AliasBasedTokenStateService.java

```diff
@@ -446,13 +446,15 @@ protected void addMetadataInMemory(String tokenId, TokenMetadata metadata) {
   }
 
   @Override
-  public TokenMetadata getTokenMetadata(String tokenId) {
+  public TokenMetadata getTokenMetadata(String tokenId) throws UnknownTokenException {
     TokenMetadata tokenMetadata = super.getTokenMetadata(tokenId);
```

Comment:
UnknownTokenException should be handled. In HA case it may happen that the in-memory collection will not have the given token metadata -> super will throw UTE. We should catch it and continue with the alias service action.

Reply:
Do you think we need to log the fact that it's missing from the in-memory record? I think not, but I'm interested in your opinion.